File: src/main/java/com/example/backend/model/Projects/UserProjectRequest.java
package com.example.backend.model.Projects;

import com.example.backend.ToExport;
import com.example.backend.model.Users.Users;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;


/**
 * DTO för userproject - denna klass används endast för att hantera den användardata som är relevant inom ett projekt och endast inom
 * ett projekt. En blueprint för en användare inom ett projekt.
 */
@ToExport
@Getter
@Setter
public class UserProjectRequest {

    @Size(max = 15)
    private String role;
    private boolean isAdmin;
    private boolean hasJoined;
    private RequestType requestType;
}

File: app/src/main/java/com/example/frontend_android/model/Projects/UserProjectRequest.kt
package com.example.frontend_android.model.Projects

import com.example.frontend_android.test.requestType


/**
 * DTO för projekt
 */
data class UserProjectRequest (
    val userId: Long,
    val projectId: Long,
    private val role: String? = null,
    private val isAdmin: Boolean? = null,
    private val joined: Boolean? = null,
    private val requestType: String
)